File: katas/7 kyu/youre-a-square/solution.py
# youre-a-square
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def is_square(n):    
  return True if (n**0.5)**2 == n else False
# ------------------------------ CLEVER SOLUTION ----------------------------- #
# ----------------------------------- TEST ----------------------------------- #
test.assert_equals(is_square(-1), False, "-1: Negative numbers cannot be square numbers")
test.assert_equals(is_square( 0), True, "0 is a square number (0 * 0)")
test.assert_equals(is_square( 3), False, "3 is not a square number")
test.assert_equals(is_square( 4), True, "4 is a square number (2 * 2)")
test.assert_equals(is_square(25), True, "25 is a square number (5 * 5)")
test.assert_equals(is_square(26), False, "26 is not a square number")

File: katas/7 kyu/youre-a-square/solution.js
// youre-a-square
/* ------------------------------- MY SOLUTION ------------------------------ */
var isSquare = function(n){
  return Math.floor(n ** 0.5) ** 2 == n;
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
const chai = require("chai");
const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
chai.config.truncateThreshold=0;

describe("isSquare", function(){
  it("should work for some examples", function(){
    Test(isSquare(-1), false, "-1: Negative numbers cannot be square numbers");
    Test(isSquare( 0), true, "0 is a square number (0 * 0)");
    Test(isSquare( 3), false, "3 is not a square number");
    Test(isSquare( 4), true, "4 is a square number (2 * 2)");
    Test(isSquare(25), true, "25 is a square number (5 * 5)");
    Test(isSquare(21), false, "26 is not a square number");
  });
});